src/test/java/gov/healthit/chpl/aqa/asserts/ViewProductPageAsserts.java

```diff
@@ -0,0 +1,70 @@
+package gov.healthit.chpl.aqa.asserts;
+
+import static org.junit.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
+
+import org.openqa.selenium.JavascriptExecutor;
+
+import cucumber.api.java.en.Then;
+import gov.healthit.chpl.aqa.pageObjects.ViewProductPage;
+import gov.healthit.chpl.aqa.stepDefinitions.Base;
+
+/**
+ * Class ViewProductPageAsserts definition.
+ */
+public class ViewProductPageAsserts extends Base {
+
+    /**
+     * Assert that product edit link exists on view version page.
+     * @param productId is the product id passed in page objects
+     */
+    @Then("^I see product edit link for product \"([^\"]*)\"$")
+    public void iSeeProductEditLink(final String productId) {
+        ViewProductPage.editProductLink(getDriver(), productId).isDisplayed();
+        assertTrue(true);
+    }
+
+    /**
+     * Assert that product split link exists on view version page.
+     * @param productId is the product id passed in page objects
+     */
+    @Then("^I see product split link for product \"([^\"]*)\"$")
+    public void iSeeProductSplitLink(final String productId) {
+        ViewProductPage.splitProductLink(getDriver(), productId).isDisplayed();
+        assertTrue(true);
+    }
+
+    /**
+     * Assert that product merge link exists on view version page.
+     * @param productId is the product id passed in page objects
+     */
+    @Then("^I see product merge link for product \"([^\"]*)\"$")
+    public void iSeeProductMergeLink(final String productId) {
+        ViewProductPage.mergeProductLink(getDriver(), productId).isDisplayed();
+        assertTrue(true);
```

Comment:
This was something you'd changed in the Version page, right? Should do the same thing here.

`assertTrue(true)` will always pass

Same with the other three

Reply:
Nope, I verified the text in the version feature whereas for the link it's the same as above!

Reply:
So if you edit the scenario for the merge to include ROLE_ACB, and run the test. Does it pass or fail?

Reply:
It fails!

Reply:
Does it fail because of a stackdump, or does it write out something like "expected true, was false"?

Reply:
It throws an exception and does not write anything like "expected true...."

Reply:
If the exception is on the line above, then that makes sense. Taking an exception as a "good thing" to indicate a test that doesn't pass isn't good practice. That means your code isn't getting to the `assertTrue` line, and that means you're not actually testing what you want to test.

See first paragraph in this answer: https://stackoverflow.com/a/2496577/16593 It's a java based thing, but the differentiation between test **Failure** and test **Exception** is the important thing